demo/nvrtc_test.cpp

```diff
@@ -0,0 +1,89 @@
+#include "Components.h"
+#include "ufcx.h"
+#include "nvrtc.h"
+#include <stdexcept>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+int main()
+{
+  // extract kernel
+  ufcx_integral* integral = form_Components_L->form_integrals[0];
+  ufcx_tabulate_tensor_cuda_nvrtc* kernel = integral->tabulate_tensor_cuda_nvrtc;
+  // call kernel to get CUDA-wrapped source code
+  int num_program_headers;
+  const char** program_headers;
+  const char** program_include_names;
+  const char* program_src;
+  const char* tabulate_tensor_function_name;
+  if (!kernel) {
+    throw std::runtime_error("NVRTC wrapper function is NULL!");
+  }
+  (*kernel)(
+    &num_program_headers, &program_headers,
+    &program_include_names, &program_src,
+    &tabulate_tensor_function_name);
+  // compile CUDA-wrapped source code with NVRTC
+  // with proper error checking
+
+  nvrtcResult nvrtc_err;
+  nvrtcProgram program;
+  nvrtc_err = nvrtcCreateProgram(
+    &program, program_src, tabulate_tensor_function_name,
+    num_program_headers, program_headers,
+    program_include_names);
+
+ if (nvrtc_err != NVRTC_SUCCESS) {
+    throw std::runtime_error(
+      "nvrtcCreateProgram() failed with " +
+      std::string(nvrtcGetErrorString(nvrtc_err)) + " "
+      "at " + std::string(__FILE__) + ":" + std::to_string(__LINE__));
+  }
+
+  int num_compile_options = 0;
+  const char** compile_options;
+  // Compile the CUDA C++ program
+  nvrtcResult nvrtc_compile_err = nvrtcCompileProgram(
+    program, num_compile_options, compile_options);
+  if (nvrtc_compile_err != NVRTC_SUCCESS) {
+    // If the compiler failed, obtain the compiler log
+    std::string program_log;
+    size_t log_size;
+    nvrtc_err = nvrtcGetProgramLogSize(program, &log_size);
+    if (nvrtc_err != NVRTC_SUCCESS) {
+      program_log = std::string(
+        "nvrtcGetProgramLogSize() failed with " +
+        std::string(nvrtcGetErrorString(nvrtc_err)) + " "
+        "at " + std::string(__FILE__) + ":" + std::to_string(__LINE__));
+    } else {
+      program_log.resize(log_size);
+      nvrtc_err = nvrtcGetProgramLog(
+        program, const_cast<char*>(program_log.c_str()));
+      if (nvrtc_err != NVRTC_SUCCESS) {
+        program_log = std::string(
+          "nvrtcGetProgramLog() failed with " +
+          std::string(nvrtcGetErrorString(nvrtc_err))) + " "
+          "at " + std::string(__FILE__) + ":" + std::to_string(__LINE__);
+      }
+      if (log_size > 0)
+        program_log.resize(log_size-1);
+    }
+    nvrtcDestroyProgram(&program);
+
+    std::stringstream ss;
+    ss << "nvrtcCompileProgram() failed with "
```

Comment:
You can use std::format if you can switch to C++20, removes the need for using C++ terrible string formatting.

Reply:
So I tried switching this code to use std::format, however the C++ compiler in Github's CI environment appears to not support c++20 (tests worked locally, but broke during CI on Github).